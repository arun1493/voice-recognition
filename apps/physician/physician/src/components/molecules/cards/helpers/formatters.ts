import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
