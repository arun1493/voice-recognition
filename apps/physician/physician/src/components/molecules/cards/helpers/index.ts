export * from "./labels";
export * from "./formatters";
