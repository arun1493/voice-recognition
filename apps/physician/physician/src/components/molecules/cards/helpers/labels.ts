import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
