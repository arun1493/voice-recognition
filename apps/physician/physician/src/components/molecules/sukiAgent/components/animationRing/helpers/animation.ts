import { RING_ANIMATION_DURATION, RING_BASE_OPACITY, RING_SCALE, SukiAgentState } from "../../../../../../constants";

function isValidSukiAgentStateForSingleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (isAgentActive && agentStatus === SukiAgentState.AWAKE) || agentStatus === SukiAgentState.PROCESSING;
}

function isValidSukiAgentStateForMultipleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (
    isAgentActive &&
    !isValidSukiAgentStateForSingleRingAnimation(agentStatus, isAgentActive) &&
    agentStatus === SukiAgentState.LISTENING
  );
}

function getDictationRingVariants(ringsCount: number) {
  const dictationRingVariants = [];
  for (let i = ringsCount; i > 0; i--) {
    const RING_VARIANTS = {
      oscillate: {
        opacity: RING_BASE_OPACITY - i * 0.2,
        scale: RING_SCALE + i * 0.01,
        transition: {
          yoyo: Infinity,
          duration: RING_ANIMATION_DURATION,
          type: "tween",
          delay: i * 0.1,
          ease: "easeInOut"
        }
      }
    };

    dictationRingVariants.push(RING_VARIANTS);
  }
  return dictationRingVariants;
}

export {
  getDictationRingVariants,
  isValidSukiAgentStateForMultipleRingAnimation,
  isValidSukiAgentStateForSingleRingAnimation
};
