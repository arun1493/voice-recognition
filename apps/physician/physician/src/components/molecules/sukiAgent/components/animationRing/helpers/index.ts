export * from "./ringSize";
export * from "./animation";
