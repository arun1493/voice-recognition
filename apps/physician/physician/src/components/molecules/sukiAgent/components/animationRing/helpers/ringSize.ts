import { SukiAgentRingWidth, SukiAgentSize } from "../../../../../../constants";

function getRingWidth(agentSize: SukiAgentSize, agentPosition: "center" | "bottom") {
  if (agentPosition === "bottom") return SukiAgentRingWidth[SukiAgentSize.SM];
  return SukiAgentRingWidth[agentSize] || SukiAgentRingWidth[SukiAgentSize.XL];
}

export { getRingWidth };
