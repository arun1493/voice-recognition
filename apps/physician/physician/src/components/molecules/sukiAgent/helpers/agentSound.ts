import { logger } from "../../../../api";
import { AgentSoundPaths, AgentSounds } from "../constants";

function playSound(agentSound: AgentSounds) {
  if (window.Audio) {
    const soundFilePath = AgentSoundPaths[agentSound];
    const audio = new Audio(soundFilePath);
    const promise = audio.play();
    if (promise !== undefined) {
      promise.catch((error) => {
        logger.error({ message: "Audio Error", payload: error });
      });
    }
  }
}

export { playSound };
