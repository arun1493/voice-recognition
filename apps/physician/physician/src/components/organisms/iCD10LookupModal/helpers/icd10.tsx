import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
