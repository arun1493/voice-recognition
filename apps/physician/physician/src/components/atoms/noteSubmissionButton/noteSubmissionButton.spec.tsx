import { useState } from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup, fireEvent, render } from "@testing-library/react";

import { NoteSubmissionButton } from "./";
import { configureI18n } from "../../../helpers";
import { shouldMatchSnapshot } from "../../../helpers/snapshot.spec";
import { componentSetupWithStore, componentWithTheme } from "../../../hocs";

const setup = componentSetupWithStore(() => <NoteSubmissionButton onClick={() => null} size="sm" />);

describe("NoteSubmissionButton", () => {
  configureI18n();
  let baseElement: Element;

  beforeEach(() => {
    baseElement = setup().baseElement;
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    expect(baseElement).toBeTruthy();
  });

  it("should have I'm done text and svg", () => {
    expect(baseElement).toHaveTextContent("I'm done");
    const svg = baseElement.getElementsByTagName("svg");
    expect(svg).toHaveLength(1);
  });

  it("should match snapshot", () => {
    const props = {
      size: "sm",
      onClick: () => null
    };

    shouldMatchSnapshot(NoteSubmissionButton, props);
  });

  it("should respond to clicks", () => {
    // changing size of component on click
    const TestComponent = () => {
      const [size, setSize] = useState("xs");

      return componentWithTheme(NoteSubmissionButton, {
        size,
        onClick: () => {
          setSize("sm");
        }
      });
    };

    const { container, baseElement } = render(<TestComponent />);

    const getHeight = (element: Element) => getComputedStyle(element).getPropertyValue("height");

    const element = baseElement?.lastChild?.lastChild as Element;

    expect(getHeight(element)).toBe("40px");

    fireEvent.click(container.getElementsByTagName("button")[0]);
    // size became "sm"
    expect(getHeight(element)).toBe("40px");
  });

  it("should not respond to clicks if disabled", () => {
    // trying to change size of component on click when disabled flag passed
    const TestComponent = () => {
      const [size, setSize] = useState("xs");

      return componentWithTheme(NoteSubmissionButton, {
        size,
        disabled: true,
        onClick: () => {
          setSize("sm");
        }
      });
    };

    const { container, baseElement } = render(<TestComponent />);

    const getHeight = (element: Element) => getComputedStyle(element).getPropertyValue("height");

    const element = baseElement?.lastChild?.lastChild as Element;

    expect(getHeight(element)).toBe("40px");

    fireEvent.click(container.getElementsByTagName("button")[0]);
    // nothing changed
    expect(getHeight(element)).toBe("40px");
  });
});
