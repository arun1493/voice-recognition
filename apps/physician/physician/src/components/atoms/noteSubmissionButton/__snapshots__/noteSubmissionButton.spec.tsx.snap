// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteSubmissionButton should match snapshot 1`] = `
<button
  className="buttonstyled__StyledButton-ws70ng-0 tGFSu noteSubmissionButtonstyled__StyledNoteSubmissionButton-uuu5z1-1 fUMMgh"
  data-cy="button"
  disabled={false}
  onClick={[Function]}
  size="sm"
  type="button"
>
  <svg
    fill="none"
    height={18}
    viewBox="0 0 15 11"
    width={18}
  >
    <path
      d="M12.0392 0.366117C12.5274 -0.122038 13.3188 -0.122039 13.807 0.366116C14.2951 0.854272 14.2951 1.64573 13.807 2.13388L5.64431 10.2966C5.15615 10.7847 4.36469 10.7847 3.87654 10.2966L0.366116 6.78614C-0.122039 6.29799 -0.122038 5.50653 0.366116 5.01837C0.854272 4.53022 1.64573 4.53022 2.13388 5.01837L4.76043 7.6449L12.0392 0.366117Z"
      fill="#2e5f62"
    />
  </svg>
  <p
    className="textstyled__StyledText-sc-1vedkxi-0 gQoCmW noteSubmissionButtonstyled__StyledText-uuu5z1-0 dyEzra"
    color="inherit"
    cursor="default"
    onClick={[Function]}
    size="md"
  >
    I'm done
  </p>
</button>
`;
